devito/types/basic.py

```diff
@@ -497,8 +497,24 @@ def _new(cls, *args, **kwargs):
                 # Constructor if input is list of list as (row, cols, list_of_list)
                 # doesn't work as it expects a flattened.
                 newobj = super(AbstractTensor, cls)._new(args[2])
+
+            # Filter grid and dimensions
+            grid = {getattr(c, 'grid', None) for c in newobj._mat} - {None}
```

Comment:
when we gather objects, the convention in devito is to use plural names for variables, ie `grids`. It's when you reduce to a single grid, eg through `grids.pop()`, that you assign it to `grid`. But until this point you need to use `grids` as, conceptually, you still don't know how many there are

Reply:
True, typo, fixed.